bundles/specmate-integration-test/src/com/specmate/test/integration/support/DummyProjectService.java

```diff
@@ -21,6 +23,13 @@ public IProject getProject(String projectName) {
 		return projects.get(projectName);
 	}
 	
+	@Override
+	public List<String> getProjectNames() {
+		List<String> names = new ArrayList<>();
```

Comment:
instead of creating a new list every time, you could just use Set as return type and return directly the key set (may be a readonly view --> see Collections.asUnmodifiableSet)

Reply:
Yes, I changed that in the ProjectService API. However, the REST service still creates and returns a list since sets do not have a default marshal implementation in javax.